packages/dev/core/src/scene.ts

```diff
@@ -1130,7 +1130,11 @@ export class Scene implements IAnimatable, IClipPlanesHolder, IAssetContainer {
      * @returns the computed eye position
      */
     public bindEyePosition(effect: Nullable<Effect>, variableName = "vEyePosition", isVector3 = false): Vector4 {
-        const eyePosition = this._forcedViewPosition ? this._forcedViewPosition : this._mirroredCameraPosition ? this._mirroredCameraPosition : this.activeCamera!.globalPosition;
+        const eyePosition = this._forcedViewPosition
+            ? this._forcedViewPosition
+            : this._mirroredCameraPosition
+              ? this._mirroredCameraPosition
+              : (this.activeCamera?.globalPosition ?? Vector3.ZeroReadOnly);

```

Comment:
There's no default active camera anymore in the new "frame graph path".

The `scene.activeCamera` property is now only set when the underlying code needs it (like when rendering meshes with the `ObjectRenderer`), but is reset to `null` afterwards. A shadow generator doesn't need a camera (we only need a list of meshes + a light to generate a shadow map - at least in the non-CSM shadow generator case), so `scene.activeCamera` is null during this process. As `bindEyePosition` is always called as part of material management, it would fail without the check on `this.activeCamera`.

The right fix would be to be able to bypass the call to `bindEyePosition` when rendering to a shadow map, but I don't see how to do this cleanly...